lib/r10k/api/modules.rb

```diff
@@ -123,12 +125,19 @@ def write_module(module_name, env_map, path, opts={})
             git.clean(path, git_dir: git_dir, force: true)
           end
         when :forge
-          tarball_cachedir = cachedir_for_forge_module(mod[:source], opts[:cachedir])
+          release_slug = release_slug_from_module_slug_version(mod[:source], mod[:resolved_version])
+          cachedir = opts[:cachedir]
 
-          # TODO: cache tarball
+          unpack_tmpdir = cachedir_for_forge_module([release_slug, 'unpack'].join('-'), cachedir)
+
+          release_tarball = get_cached_forge_release(release_slug, cachedir, false, opts[:forge])
+
+          begin
+            Dir.mktmpdir(release_slug, unpack_tmpdir) do |tmpdir|
```

Comment:
The docs for `Dir.mktmpdir` indicate that the block form will remove the tmpdir before returning:

```
If a block is given, it is yielded with the path of the directory. The
directory and its contents are removed using FileUtils.remove_entry before
Dir.mktmpdir returns. The value of the block is returned.
```

Is this using a tmpdir to unpack and then copying the contents from the tmpdir to the final location?